File: app/src/main/java/closer/vlllage/com/closer/handler/map/MapZoomHandler.java
package closer.vlllage.com.closer.handler.map;

import closer.vlllage.com.closer.pool.PoolMember;
import io.reactivex.Observable;
import io.reactivex.subjects.BehaviorSubject;

public class MapZoomHandler extends PoolMember {

    private BehaviorSubject<Float> onZoomChanged = BehaviorSubject.create();

    public void update(float zoom) {
        onZoomChanged.onNext(zoom);
    }

    public Observable<Boolean> onZoomGreaterThanChanged(final float zoomGreaterThan) {
        return onZoomChanged.map(zoom -> zoom >= zoomGreaterThan).distinctUntilChanged();
    }
}

File: app/src/main/java/closer/vlllage/com/closer/handler/map/MapZoomHandler.kt
package closer.vlllage.com.closer.handler.map

import com.queatz.on.On
import io.reactivex.rxjava3.core.Observable
import io.reactivex.rxjava3.subjects.BehaviorSubject

class MapZoomHandler constructor(private val on: On) {

    private val onZoomChanged = BehaviorSubject.create<Float>()

    fun update(zoom: Float) {
        onZoomChanged.onNext(zoom)
    }

    fun onZoomGreaterThanChanged(zoomGreaterThan: Float): Observable<Boolean> {
        return onZoomChanged.map { zoom -> zoom >= zoomGreaterThan }.distinctUntilChanged()
    }
}
